### ISSUE TITLE
[Question] Should we switch logrus to uber/zap

### ISSUE THREAD

**Issue by denis-tingajkin** [networkservicemesh/sdk#418]
## Description

zap is Blazing fast, structured, leveled logging in Go. https://github.com/uber-go/zap 

It has better performance than logrus ~30x times 
https://github.com/uber-go/zap#performance

Question: Should we try to replace logrus to zap?

**Comment by denis-tingajkin** [comment]
@haiodo , @edwarnicke , @fkautz ^^^

**Comment by edwarnicke** [comment]
So is the motivation performance?  Does it support richer semantics like logrus does (adding tags to entries etc)?  Usually in these conversations its a tradeoff between perf and expressiveness of logs.

**Comment by denis-tingajkin** [comment]
@edwarnicke

I think motivation performance actually can make sense, for example it can reduce the starting time of POD and make applications more responsive. 

>  Does it support richer semantics like logrus does (adding tags to entries etc)?

I've quickly looked and locally tested, looks like all logrus functionality exist in zap. They're also a lot of things for logging costumization.

Also, an additional benefit of using zap than logrus is less memory allocation (zap uses less memory then logrus in >= ~50 times).

And I've found that jaeger has zap support https://godoc.org/github.com/jaegertracing/jaeger-client-go/log/zap

**Comment by edwarnicke** [comment]
@denis-tingajkin When I looked at zap... I didn't see the ability to add tags, or hooks... did I miss that?

As to memory and startup times... do we have a notion of the memory/latency savings to for us using zap instead of zoom?  10 microseconds out of 100 ms would likely be kind of uninteresting... but larger savings might be.  Likewise, of logrus is taking 100kb... a 50kb savings out of a 10mb memory footprint isn't that interesting... but if the memory footprint is much larger it becomes more interesting :)

**Comment by denis-tingajkin** [comment]
@edwarnicke Yes, I think you are thinking right. We should check how much memory will be reduced for some applications if we'll switch on zap. Also, we should keep a thought in mind that each difference between logrus and zap will be scaled on `tests count * pods count in test`.

**Comment by denis-tingaikin** [comment]
This is looking as a good first issue. 
 Need to do: Investigate and provide comparasion between zap and logrus in load.
